common_utils/profile.py

```diff
@@ -0,0 +1,91 @@
+from string import Template
+
+import jmespath
+import requests
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from requests.auth import AuthBase
+
+from common_utils.oidc import TunnistamoTokenExchange
+
+
+class BearerAuth(AuthBase):
+    """Bearer token authentication module for requests."""
+
+    def __init__(self, token: str):
+        self.token = token
+
+    def __call__(self, r):
+        r.headers["authorization"] = "Bearer " + self.token
+        return r
+
+
+class ProfileAPI:
+    """Client for fetching open-city-profile related data."""
+
+    timeout = 5
+
+    def __init__(self):
+        self.check_settings()
+
+    @staticmethod
+    def check_settings():
+        if not settings.HELSINKI_PROFILE_API_URL:
+            raise ImproperlyConfigured(
+                "Required Helsinki profile configuration not set."
+            )
+
+    def fetch_profile(self, authorization_code: str, id: str) -> dict:
+        """Fetch profile data for the given profile ID. Requires staff level permission"""
+        query = Template(
+            """
+            query Profile {
+                profile(id: "${id}", serviceType: ${service_type}) {
+                    id
+                }
+            }
+        """
+        ).substitute(id=id, service_type=settings.HELSINKI_PROFILE_SERVICE_TYPE)
```

Comment:
I think the preferred way would be to use GQL variables for the dynamic parts and keep the query static.